### PR TITLE
K8s versioning

### PR DESCRIPTION
**Pull request by lasiva**
There are many changes, since I have accommodated Kevin's request to rename some variables from viya_kubelet_version_min to viya_k8s_version_min.

Unit tested on AWS.

### CODE REVIEW

File: pre_install_report/README.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ Kubernetes may orchestrate once Viya is deployed.  The report and the informatio
 be considered a snapshot in time.  
 
 The Kubernetes cluster for a SAS Viya deployment must meet the requirements documented in [SAS® Viya® Operations](https://go.documentation.sas.com/doc/en/itopscdc/default/itopssr/titlepage.htm)  
-Ensure that the Kubernetes version is within the documented range for the selected cloud provider. 
+Ensure that the Kubernetes version is within the documented range for the selected cloud provider.  If the Kubernetes server version is below a current default minimum, a warning will be included in the report.
```
Comment (FredPerrySAS) [inline pre_install_report/README.md:13]: I'd recommend changing "a current default minimum" to just "the default minimum".

File: pre_install_report/README.md [hunk 2]
```diff
@@ -146,6 +146,10 @@ web-viewable, HTML format.
 You can modify the <tool-download-dir>/viya4-ark/pre_install_report/viya_deployment_settings.ini file to alter the   
 minimum and aggregate settings for CPU and memory on nodes. For more information, see the details in the file.
 
+If you modify the VIYA_K8S_VERSION_MIN to a version less than the minimum Kubernetes version supported by this 
+release of the report tool, you are operatiing outside the supported capabilities of the report tool.  It is better
```
Comment (FredPerrySAS) [inline pre_install_report/README.md:150]: "operating" misspelled

Comment (FredPerrySAS) [inline pre_install_report/README.md:150]: "It is better to use a release of Viya 4 ARK tools matching the..." should be SAS recommends using a release of Viya 4 ARK tools that matches the..."

File: pre_install_report/README.md [hunk 3]
```diff
@@ -146,6 +146,10 @@ web-viewable, HTML format.
 You can modify the <tool-download-dir>/viya4-ark/pre_install_report/viya_deployment_settings.ini file to alter the   
 minimum and aggregate settings for CPU and memory on nodes. For more information, see the details in the file.
 
+If you modify the VIYA_K8S_VERSION_MIN to a version less than the minimum Kubernetes version supported by this 
+release of the report tool, you are operatiing outside the supported capabilities of the report tool.  It is better 
+to use a release of Viya 4 ARK tools matching the required minimum you may be working with.
```
Comment (FredPerrySAS) [inline pre_install_report/README.md:151]: "you may be" should be "you are"

**Review by FredPerrySAS — CHANGES_REQUESTED** [review]
See comment inline

File: pre_install_report/README.md (hunk 1 above)
Comment (kevinlinglesas) [reply pre_install_report/README.md]: Done.

File: pre_install_report/README.md (hunk 2 above)
Comment (kevinlinglesas) [reply pre_install_report/README.md]: Done.  Thank you @FredPerrySAS .

**Comment by kevinlinglesas** [comment]
I'd also invite others to review the README file changes.   Does the content cover what we need to document at this point for this enhancement?    My approach was to add minimal changes necessary but to also include the support statement we discussed (@erharb , @lasiva ).

File: pre_install_report/library/pre_install_check.py [hunk 4]
```diff
@@ -76,21 +77,26 @@ def __init__(self, sas_logger: ViyaARKLogger, viya_kubelet_version_min,
 
     def _parse_release_info(self, release_info):
         """
-        This method checks that the format of the VIYA_KUBELET_VERSION_MIN specfied in the
-        user modifiable properies file is valid.
+        This method checks that the format of the VIYA_K8s_VERSION_MIN specified in the
```
Comment (kevinlinglesas) [inline pre_install_report/library/pre_install_check.py:80]: Nit.   Typo in variable name, lowercase 's'

Comment (lasiva) [reply pre_install_report/library/pre_install_check.py:80]: Done

File: pre_install_report/library/pre_install_check.py [hunk 5]
```diff
@@ -1055,6 +1064,9 @@ def get_calculated_aggregate_memory(self):
     def set_k8s_version(self, version: Text):
         self._k8s_server_version = version
 
+    def set_k8s_version_min(self, version: Text):
```
[the remark this answers is not in the document]
Comment (lasiva) [reply pre_install_report/library/pre_install_check.py:1067]: This setter method is used currently in the viya4-ark/pre_install_report/test/test_pre_install_report.py

File: pre_install_report/pre_install_report.py [hunk 6]
```diff
@@ -252,10 +252,18 @@ def main(argv):
...
-        sas_pre_check_report: ViyaPreInstallCheck = \
-            ViyaPreInstallCheck(sas_logger, check_limits['items']['VIYA_KUBELET_VERSION_MIN'],
-                                check_limits['items']['VIYA_MIN_AGGREGATE_WORKER_CPU_CORES'],
-                                check_limits['items']['VIYA_MIN_AGGREGATE_WORKER_MEMORY'])
+        try:
+            sas_pre_check_report: ViyaPreInstallCheck = \
+                ViyaPreInstallCheck(sas_logger, check_limits['items']['VIYA_K8S_VERSION_MIN'],
+                                    check_limits['items']['VIYA_MIN_AGGREGATE_WORKER_CPU_CORES'],
+                                    check_limits['items']['VIYA_MIN_AGGREGATE_WORKER_MEMORY'])
+        except KeyError as e:
```
[the remark this answers is not in the document]
Comment (lasiva) [reply pre_install_report/pre_install_report.py:260]: Yes, I found this oversight and removed it.

[the remark this answers is not in the document]
Comment (lasiva) [reply pre_install_report/pre_install_report.py:260]: Yes, When I found it, I cleaned it up.

**Review by kevinlinglesas — APPROVED** [review]
lgtm

**Pull request merged**